prudp_endpoint.go

```
@@ -285,7 +289,7 @@ func (pep *PRUDPEndPoint) handleConnect(packet PRUDPPacketInterface) {
 
 	payload := make([]byte, 0)
 
-	if pep.IsSecureEndpoint {
+	if len(packet.Payload()) != 0 {
```

If we only check the payload someone could login to the server without an account, which we don't want. we could check if `AccountDetailsByPID` and `AccountDetailsByUsername` aren't null for this case

---

What happens on the real servers when this happens, do we know?

And how would this allow for someone to login to an account server if they don't have a NEX account?

I do agree with better checks though, I was just trying to find a way to remove one-off config stuff. I'm not sure  about the `AccountDetailsByPID` and `AccountDetailsByUsername` checks though? How would that determine if the server is the secure server, since those should be set on all servers?

---

> What happens on the real servers when this happens, do we know?

As per my tests on the NEX 1 server, if the secure server receives a CONNECT packet with no payload or invalid, it just won't send anything back and silently fail.

> And how would this allow for someone to login to an account server if they don't have a NEX account?

That's the thing, it won't technically have an account and it would essentially behave like if someone used the `guest` account, which could break things.

> How would that determine if the server is the secure server, since those should be set on all servers?

You are right, I didn't see that when I first sent the review. In that case I don't see any better option than to bring the `IsSecureEndpoint` option back.

---

> As per my tests on the NEX 1 server, if the secure server receives a CONNECT packet with no payload or invalid, it just won't send anything back and silently fail.

Sounds good, I'm fine with just silenting failing

> That's the thing, it won't technically have an account and it would essentially behave like if someone used the `guest` account, which could break things.

Would it not just fail to connect?

> You are right, I didn't see that when I first sent the review. In that case I don't see any better option than to bring the `IsSecureEndpoint` option back.

That's fine. Honestly this might be the way NEX does it too. Given that the secure server just silently fails and does nothing, not even acting like an authentication server (sending back the CONNECT response and such), it seems like those servers really do have configuration differences beyond how they're interacted with

I think adding this option back is the right to go as well